src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2773,11 +2774,7 @@ object Desugar extends LazyLogging {
             dop <- go(op)
           } yield dop match {
             case dop : in.ArrayLit => in.IntLit(dop.length)(src)
```

Comment:
Is there a reason for specialcasing `ArrayLit` here? Given that you've extended the array encoding to support `cap`, it seems to me we could simply convert `PCapacity` to `Capacity`.
Furthermore, the same comment also applies to the lines above for the `PLength` case

Reply:
good point, I don't see immediately why you would do that, except for wanting to avoid a call to the length function, which may be useful in some cases to avoid introducing triggering terms (for the same reason, it may lead to confusing triggering behaviour, as a pattern that is expected to occur does not occur in the generated viper program). I have simplified this

Reply:
ah, this special case was prior to my changes. I removed the special case

Reply:
Yes, I could have been clearer on this ^^